src/registry/docker/docker.go

```diff
@@ -83,12 +94,48 @@ func dockerContentDigest(req *http.Request) ([]string, error) {
 
 	digest, found := resp.Header["Docker-Content-Digest"]
 	if !found {
-		return nil, fmt.Errorf("docker API did not return a content digest for this image")
+		// OCI images will not respond with the list of digest
+		return nil, nil
 	}
 
 	return digest, nil
 }
 
+func ociDigests(req *http.Request) ([]string, error) {
+	// https://github.com/moby/buildkit/issues/2251
+	acceptHeader := "application/vnd.oci.image.index.v1+json"
+	req.Header.Set("Accept", "application/vnd.oci.image.index.v1+json")
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("querying dockerhub tags endpoint: %w", err)
+	}
+
+	defer resp.Body.Close()
+
+	if resp.Header.Get("Content-Type") != acceptHeader {
+		return nil, nil
+	}
```

Comment:
Seems like this should be an error or at least a log, wouldn't it?

Reply:
the api return different content-type event if the accept header is set to the image one. this was a quick way to filter this responses, but I didn't want to log error since this is expected and logging this will pollute the logs.